### PR TITLE
Lint document and add linting step

### PR DESCRIPTION
**Pull request by tyler36**
This PR does some minor cleanup:

- standardize indenting
- remove duplicate word

### CODE REVIEW

**Review by rfay — APPROVED** [review]
Looks good to me. Do you have an automated way of linting that you could do a PR for?

**Review by mattstein — APPROVED** [review]

**Comment by tyler36** [comment]
I manually updated it. If I find a github action, I'll make another PR.

**Comment by rfay** [comment]
I really apologize about this being all conflicted now because the other one got in there before you. If you can find a way to automate it this time even if you have to do run it manually, that may be a step forward. 

Note that Goland did *not* complain about the existing format.

**Comment by tyler36** [comment]
I have added a "prettier" action. I set the action to lint JSON, as well as JS and MD files.

Prettier is considered opinionated so you may want to change the options. (https://prettier.io/docs/en/configuration.html)
Requesting re-review as not sure if the rules conflict with what @mattstein is doing/planning for docs.

File: .github/workflows/linting.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          # Make sure the actual branch is checked out when running on pull requests
+          ref: ${{ github.head_ref }}
+
+      # Prettier Action: https://github.com/marketplace/actions/prettier-action
+      # Prettier config: https://prettier.io/docs/en/configuration.html
+      - name: Prettify code
+        uses: creyD/prettier_action@v4.3
+        with:
+          # This part is also where you can pass other options, for example:
+          prettier_options: --write **/*.{js,json,md}
```
Comment (rfay) [inline .github/workflows/linting.yaml]: --write for an action makes me a little uncomfortable, and of course it doesn't work on forked PRs (not a powerful enough GITHUB_TOKEN). So how does that work out?

**Review by rfay — COMMENTED** [review]
Please add instructions to the README about how to manually prettify. Thanks!

File: .github/workflows/linting.yaml (hunk 1 above)
Comment (rfay) [reply .github/workflows/linting.yaml]: I think the failure is a result of that, https://github.com/ddev/remote-config/actions/runs/6243383462/job/16948758817?pr=14

**Comment by mattstein** [comment]
I turn Prettier off in VS Code when I’m editing Markdown because it’s too opinionated and bossy. We’ve been here before with a quick auto-formatting change wrecking intentional line breaks and adding confusion, and I’m not keen on applying this everywhere to see what breaks. (With the write option enabled, there’d also be no way to fix it!)

That said, if everybody else is a fan go for it!

**Comment by rfay** [comment]
I don't think the commit is going to work at all for forked PRs, so perhaps just providing the way to reformat is the way?

@mattstein do you object to the actual results of prettier on this json file?

**Comment by mattstein** [comment]
No objection at all @rfay!

**Comment by rfay** [comment]
@tyler36 I'm pretty sure the best path is to change the prettier formatting to a *check* instead of a *commit*, and then just require people to have it in that format, so the README can say how.

File: .github/workflows/linting.yaml (hunk 1 above)
Comment (tyler36) [reply .github/workflows/linting.yaml]: I've update the workflow.

**Comment by tyler36** [comment]
Instructions added

**Comment by tyler36** [comment]
If we target `jsonc` files, we should probably target `json` as well.
An earlier commit updated JavaScript files, so I've added that to the workflow and manual run.

> I turn Prettier off in VS Code when I’m editing Markdown

Currently, NOT targetting markdown.

**Review by rfay — COMMENTED** [review]
I don't understand really why anything matters besides the remote-config.jsonc, and don't understand why this touches that file if it was already properly formatted.

File: README.md [hunk 2]
```diff
@@ -2,3 +2,23 @@
...
+You can manually run Prettier on this repository (requires Node.js).
+
+1. Clone or fork this project locally.
+1. Install node dependencies:
+
+   ```shell
+   npm install
+   ```
+
+1. Run the following command to attempt to auto-fix issues:
```
Comment (mattstein) [inline README.md]: I realize this is being picky, but if Markdown’s goal is to be human-readable it’s worth having sequential numbers out of principle:

```suggestion
1. Clone or fork this project locally.
2. Install node dependencies:

   ```shell
   npm install
   ```

3. Run the following command to attempt to auto-fix issues:
```

**Review by mattstein — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (rfay) [reply README.md]: > if Markdown’s goal is to be human-readable it’s worth having sequential numbers out of principle

Somebody actually agrees with me! I never thought anybody did!

Comment (mattstein) [reply README.md]: @rfay [John Gruber agrees with you](https://daringfireball.net/projects/markdown/), which should count for a lot.

> The overriding design goal for Markdown’s formatting syntax is to make it as readable as possible. The idea is that a Markdown-formatted document should be publishable as-is, as plain text, without looking like it’s been marked up with tags or formatting instructions.

Comment (rfay) [reply README.md]: It sure makes maintaining markdown harder. But... you should be able to read it!

Comment (tyler36) [reply README.md]: > I don't understand really why anything matters besides the remote-config.jsonc

> I turn Prettier off in VS Code when I’m editing Markdown

I'm lost. You want to include markdown files in the linting process now?

Comment (rfay) [reply README.md]: Sorry. I guess Matt was responding to earlier version, didn't realize that whole section had been removed, and then we went off on a tangent about markdown numbers. Nothing is wrong.

Comment (rfay) [reply README.md]: Nothing is wrong except that Matt was saying use 1, 2, 3 in the README instead of 1, 1, 1. Not about linting.

Comment (mattstein) [reply README.md]: Oh yikes, sorry if I pointed at obsolete content! I _thought_ I was reviewing Markdown that had been added to the readme—which as you said had nothing to do with the linting itself.

**Comment by rfay** [comment]
This does seem to have trouble @tyler36 - it doesn't pass its own tests `A branch or tag with the name '2023092_linting' could not be found` - is it unable to do forked PRs, or ... ???

**Comment by rfay** [comment]
Yay!

**Comment by tyler36** [comment]
Anything else?

**Comment by bmartinez287** [comment]
We may want to grab the latest version of remote-config.jsonc. Otherwise, some of the tips will be deleted.

**Comment by tyler36** [comment]
@bmartinez287 

Thanks for the heads up. File was synced with current tips.

File: README.md [hunk 3]
```diff
@@ -2,3 +2,23 @@
 
 This repository contains the DDEV remote config definitions used by DDEV. For
 more information, see [DDEV's documentation](https://ddev.readthedocs.io/en/latest/developers/remote-config).
+
+## Contributing
+
+### Style guide
```
Comment (mattstein) [inline README.md]: Title case:

```suggestion
### Style Guide
```

File: README.md [hunk 4]
```diff
@@ -2,3 +2,23 @@
...
+
+This repository uses [Prettier](https://prettier.io/docs/en/) to maintain consistent formatting.
+You can manually run Prettier on this repository (requires Node.js).
+
+1. Clone or fork this project locally.
+1. Install node dependencies:
```
Comment (mattstein) [inline README.md]:
```suggestion
2. Install node dependencies:
```

File: README.md (hunk 2 above)
Comment (mattstein) [inline README.md]:
```suggestion
3. Run the following command to attempt to auto-fix issues:
```

**Review by mattstein — CHANGES_REQUESTED** [review]

**Review by mattstein — APPROVED** [review]

**Comment by rfay** [comment]
Thanks for all the hard work on this @tyler36 .

**Pull request merged**